src/queueloader.cpp

```diff
@@ -98,166 +98,170 @@ nonstd::optional<QueueLoader::CategorizedDownloads> QueueLoader::categorize_down
-							F_OK) == 0) {
-						LOG(Level::INFO,
-							"QueueLoader::reload: "
-							"found `%s' on file "
-							"system "
-							"-> mark as partially "
-							"downloaded",
-							fn +
-							ConfigContainer::
-							PARTIAL_FILE_SUFFIX);
+						"QueueLoader::reload: "
+						"found `%s' on file "
+						"system "
+						"-> mark as already "
+						"downloaded",
+						fn);
+					if (fields.size() >= 3) {
+						if (fields[2] ==
+							"downloaded")
+							d.set_status(
+								DlStatus::
+								READY);
+						if (fields[2] ==
+							"played")
+							d.set_status(
+								DlStatus::
+								PLAYED);
+						if (fields[2] ==
+							"finished")
+							d.set_status(
+								DlStatus::
+								FINISHED);
+					} else
 						d.set_status(DlStatus::
-							ALREADY_DOWNLOADED);
-					}
-
-					d.set_url(fields[0]);
-					downloads.to_keep.push_back(d);
+							ALREADY_DOWNLOADED); // TODO: scrap DlStatus::ALREADY_DOWNLOADED state
+				} else if (
+					access((fn +
+							ConfigContainer::
+							PARTIAL_FILE_SUFFIX)
+						.c_str(),
+						F_OK) == 0) {
+					LOG(Level::INFO,
+						"QueueLoader::reload: "
+						"found `%s' on file "
+						"system "
+						"-> mark as partially "
+						"downloaded",
+						fn +
+						ConfigContainer::
+						PARTIAL_FILE_SUFFIX);
+					d.set_status(DlStatus::
+						ALREADY_DOWNLOADED);
 				}
+
+				d.set_url(fields[0]);
+				downloads.to_keep.push_back(d);
 			}
-		} while (!f.eof());
-		f.close();
-	}
+		}
+	} while (!f.eof());
 }
 
 void QueueLoader::write_queue_file(const CategorizedDownloads& downloads) const
 {
 	std::fstream f(queuefile, std::fstream::out);
-	if (f.is_open()) {
-		for (const auto& dl : downloads.to_keep) {
-			f << dl.url() << " " << utils::quote(dl.filename());
-			if (dl.status() == DlStatus::READY) {
-				f << " downloaded";
-			}
-			if (dl.status() == DlStatus::PLAYED) {
-				f << " played";
-			}
-			if (dl.status() == DlStatus::FINISHED) {
-				f << " finished";
-			}
-			f << std::endl;
+	if (!f.is_open()) {
+		return;
+	}
+
+	for (const auto& dl : downloads.to_keep) {
+		f << dl.url() << " " << utils::quote(dl.filename());
+		if (dl.status() == DlStatus::READY) {
+			f << " downloaded";
+		}
+		if (dl.status() == DlStatus::PLAYED) {
+			f << " played";
 		}
-		f.close();
+		if (dl.status() == DlStatus::FINISHED) {
+			f << " finished";
+		}
+		f << std::endl;
 	}
 }
 
 void QueueLoader::delete_played_files(const CategorizedDownloads& downloads)
 const
 {
-	if (cfg.get_configvalue_as_bool("delete-played-files")) {
-		for (const auto& dl : downloads.to_delete) {
-			const std::string filename = dl.filename();
-			LOG(Level::INFO,
-				"Deleting file %s",
-				filename);
-			if (std::remove(filename.c_str()) != 0) {
-				if (errno != ENOENT) {
-					LOG(Level::ERROR,
-						"Failed to delete file %s, error code: %d (%s)",
-						filename, errno, strerror(errno));
-				}
+	if (!cfg.get_configvalue_as_bool("delete-played-files")) {
+		return;
+	}
```

Comment:
Is there any reason for having this check in the `delete_played_files()` function instead of on the calling side?

In general, I think a check on the single calling side makes for slightly better readability, e.g.
```
if (cfg.get_configvalue_as_bool("delete-played-files")) {
    delete_played_files(…);
}
```

(although this is just my opinion, feel free to ignore)

Reply:
No particular reason, just the way it ended up. I didn't consider this when I was splitting the original, large method into the smaller ones.

I think your suggestion *is* more readable, so I applied it in https://github.com/newsboat/newsboat/pull/1426/commits/aea2231555d4d43070a67ec0a958cb1de5b0b983. Thanks!